### PR TITLE
Components: Add site selector "Add Site" popover menu with Jetpack site support

### LINKED ISSUE #6400: Jetpack NUX: Add Existing WordPress
We’ve previously discussed the idea to test having an “Add Existing WordPress” option to the site switcher.

The route for this already exists: https://wordpress.com/jetpack/connect 

What we need to do here is:
Create the UI/component needed to display a call to action to add an existing site at the bottom of the site switcher.
Link that to the route
Set this up as an a/b test on Calypso (may need help from @breezyskies team) 
 

image

### PR DESCRIPTION
**Pull request by tyxla**
### Summary

This PR fixes #6400, where it was suggested that we introduce a popover menu to the **Add New WordPress** button in the bottom of the site selector. The main purpose of this popover menu is to introduce another way to reach the self-hosted site connection flow (Jetpack Connect).

### What this PR does

1. Moves the add site button to a separate component, paving the way for the popover menu.
2. Introduces the popover menu to the new component, hooks the JPC functionality and adds some  events that allow granular tracking of the new feature.
3. Adds an A/B test that will show the new popover to only 10% of all users (cc @rickybanister for double-checking if this percentage makes sense for this feature).

### Preview

Before:

After:

### Testing instructions

1. Checkout this branch.
2. Click on "My Sites".
3. Click on "Switch Site".
4. Verify the old "Add New WordPress" button looks works just like before. 
5. Verify the new button (currently saying "Add WordPress Site") opens a popover on click with 2 options, and they work as expected:
 * __Start a new WordPress.com site__ - should lead to `/start` or `/start/survey`;
 * __Add a self-hosted WordPress site__ - should lead to `/jetpack/connect`.

### Testing notes

1. To force the A/B test to use the old button, you can type the following in your console: `localStorage.setItem('ABTests','{"siteSelectorAddSitePopover_20161011":"hidePopover"}')`

2. To force the A/B test to use the new button with popover, you can type this in the console: `localStorage.setItem('ABTests','{"siteSelectorAddSitePopover_20161011":"showPopover"}')`

### Review

/cc: 
@rickybanister, @keoshi, @meremagee for design and wording review;
@johnHackworth, @mtias for code review.

### CODE REVIEW

**Comment by keoshi** [comment]
Thanks for working on this @tyxla ! Unfortunately, I cannot get it to work by switching branches and `make run`. Also tried to `make clean` and `make distclean` before a new `make run`; no go. Is there anything else I should be doing?

**Comment by tyxla** [comment]
@keoshi that's weird. Are you getting any particular error in the console or in the terminal? If you do, it would be useful if you can share it.

Anyway, if you want to quickly test it you can do that here: https://calypso.live/?branch=add/site-selector-add-site-popover

**Comment by folletto** [comment]
> Introduces the popover menu to the new component, hooks the JPC functionality and adds some events that allow granular tracking of the new feature.

The popover should have the arrow pointing at the exact center of the `+` Gridicon. :) 

Other than that, seems working fine design wise. 👍

**Comment by keoshi** [comment]
Big thanks to @folletto for walking me through the issue. I didn't have debugging for A/B tests turned on, so it was impossible for me to see the changes.

Looks good to me!

**Comment by tyxla** [comment]
@folletto @keoshi thanks for the design review guys!

@folletto, although it makes sense for the popover arrow to be above the center of the + icon, the result would not be as good as expected - the popover menu would be cut off on the left side:



What do you think?

**Comment by folletto** [comment]
> What do you think?

That we need a third solution unfortunately. :(

We might consider the dropdown stay centered for the test, but we need to make absolutely sure it won't go in production as is if the test succeed. If that's something we can avoid, then let's proceed for the test. :) 

Alternatively, can we reposition it to be center aligned, center arrow, but aligned to the whole box, so it gets in the middle of the sidebar? I'm not sure how much workaround this involves, so "no" is an entirely acceptable answer if too complicated for the goals of the text.

**Comment by tyxla** [comment]
Center aligning the arrow and the box to the sidebar was quite easy, here's a quick preview:



I think it looks much better, but it's your call really @folletto @keoshi.

**Comment by johnHackworth** [comment]
Code and functionality-wise, this looks good to me :+1:

**Comment by keoshi** [comment]
This is a lot more balanced, looking good! :)

**Comment by johnHackworth** [comment]
Let's wait for @rickybanister to see how big the % of users we want to test this with should be and then merge!

**Comment by mtias** [comment]
I think @drw158 suggested changing the copy to "Add Site" in another thread.

**Comment by tyxla** [comment]
@mtias: personally I prefer "Add WordPress Site", but "Add Site" would also work well. Some more opinions on this are welcome.

**Comment by drw158** [comment]
It does seem a bit wordy overall. I'm not sure how many iterations the text has gone through, but it might be worth considering some more options.

The button itself I think would really benefit from a shorter CTA. I can't think of a downside of just using:

> Add Site

Maybe at one point it was just "Add WordPress". That might be fine too, but I think that's more unclear what it does.

For the popover, I thought shortening the text might work, but I can't think of a good solution. "Add a self-hosted WordPress site" is the text we use in the iOS app.

**Comment by tyxla** [comment]
Sounds good to me - I've shortened the button text to just **Add Site**. 

Regarding the popover texts - I think they're fine as they are.

**Comment by keoshi** [comment]
> For the popover, I thought shortening the text might work, but I can't think of a good solution.

We could reduce them further since the CTA is already pretty clear and the popover allows us to keep the context at all times.

>Add Site:

>- New WordPress.com site
>- Self-hosted WordPress site

_Start_ and _Add_ represent technicalities on our end and the costumer shouldn't have to know the difference. Also, if it's a WordPress.com site we're emphasizing it's a new one, while on the other option it's assumed it's an existing site for the lack thereof.

**Comment by tyxla** [comment]
@keoshi: These sound awesome too. I don't think we can shorten them anymore :)

Here is how the button and the popover look after the latest changes:

**Comment by keoshi** [comment]
I dig it — pinging @folletto @drw158 for thoughts.

**Comment by drw158** [comment]
Good idea! I like it.

**Comment by folletto** [comment]
Excellent iterations! 👍

**Comment by tyxla** [comment]
Still waiting for confirmation from @rickybanister for the ab test percent before merging.

**Comment by meremagee** [comment]
How does a typical user understand that you must _have_ a self-hosted site already to be able to _add_ it? One of these is site creation; the other is not. I think this implementation is ripe for confusion.

**Comment by folletto** [comment]
> I would much rather add a new signup step for this flow (logged in users from the add new site button) that starts off by helping guide them through a branched process, where we can support their flow with descriptive text and visuals, and where they can always go back if they've selected the "wrong" thing.

I think you're right. Since it's ready, I'd still go and test this out, and evaluate from there even if we decide to proceed with the alternative first step. :)

**Comment by meremagee** [comment]
> Since it's ready, I'd still go and test this out

Solutions that work aren't always workable solutions. This adds jargon, overhead, and complexity to already difficult-to-grok functionality. Can we at least flag this for users with at least 1 jetpack-connected site before testing?

**Comment by folletto** [comment]
>Solutions that work aren't always workable solutions. This adds jargon, overhead, and complexity to already difficult-to-grok functionality. 

Hey! That's what I usually say :D 

I still think that in this case we would get some useful information. If nobody accesses the alternative for example, or if the numbers sensibly drop, it's useful data. 

For me A/B tests aren't necessarily test meant to decide to keep the new alternative, they can be just useful to gather data. In this case, I'd go ahead to gather that data.

**Comment by meremagee** [comment]
> I still think that in this case we would get some useful information. If nobody accesses the alternative for example, or if the numbers sensibly drop, it's useful data.

This baffles me. Acknowledging that most Calypso users may not understand the nuanced differences between WordPress.com and WordPress.org, why would we assume they would know which option would better suit their needs? How can we say the clickthrough data there will be useful? The only way would be to triangulate it with qualitative data, which we could start with instead.

If "numbers sensibly drop," it may mean that real users trying to do things in Calypso have hit confusing dead ends. I'm all for testing things that we think will generally improve experiences for our users, but I can't wrap my head around shipping something confusing for the sake of seeing if some users can get around it.

**Comment by folletto** [comment]
I feel my answer would be a bit OT on the merit of this change and more on the reasons why this _kind_ of test can be useful, so I'd rather leave a proper answer to the people that initiated the request. :)

**Comment by johnHackworth** [comment]
What if we keep this format, but we make the "add a self-hosted site" option point not directly to JPC, but to a landing page where we explain what's the difference between self-hosted & wp.com, what's jetpack and why they should be using it, and a big "oh, no, I think I want a WordPress.com site" button for those who arrived there by mistake. We surely are missing a page like this.
The 'power users' who already know all this will just skip it, and for new users would be very valuable. We could also link to our pressable plan creation flow from a "I don't have a self-hosted site yet, but I want one" link there.

**Comment by rickybanister** [comment]
Hi everyone! Lots of good comments here. 

Let's take a step back a moment and evaluate what we've got—the solution proposed here served two purposes: a quick and dirty way to judge users' reactions to being able to add a site they already owned (provide an in-app path to Jetpack Connect) and to create some parity with the 'add site' solution we employ in our mobile apps. 

The two assumptions above failed to account for any confusion created by the mobile app's solution.

So here are two possible paths forward:

1. @keoshi is currently working on a much more visual solution that fortunately requires the same amount of clicks to add a Jetpack site to your account. It would add one extra click to add a WordPress.com site, but this is something we need to test. So option 1 is mothball this PR until we have designs for something with more visual explanation and fewer assumptions.

2. Launch this PR as a test only to users with 1 or more Jetpack sites already. That way we can assume that they are at least vaguely aware of the nuances between self-hosting and WordPress.com hosting. We would continue exploring the more visual 'step 1' of NUX mentioned in option 1, but perhaps gather some data about whether pre-existing users are able to successfully add self-hosted sites via the Jetpack Connect flow. 

We will share our sketches for option 1 soon!

**Comment by tyxla** [comment]
Thanks everyone for getting involved in the conversation, I appreciate that!

I'm looking forward to seeing @keoshi's work on this, and I'd love to get involved with its implementation.

But as @rickybanister stated, in case we decide to go with this PR, we should launch this for users with Jetpack sites. I've addressed this change - the popover menu will now be displayed for all users that have 1 or more Jetpack sites.

@keoshi @rickybanister: feel free to drop a message once the fate of this PR is decided. 😃

**Comment by rickybanister** [comment]
Thanks for that refactor @tyxla. I feel safe merging this and gathering some info on whether folks successfully connect additional sites. The fear of confusion has been mitigated I think.

**Comment by tyxla** [comment]
Sounds good @rickybanister, thanks. This needs a final review before merging, cc @johnHackworth  for a code review.

**Comment by johnHackworth** [comment]
I think is is ready to go ... let's ask @rickybanister  this afternoon in slack if he's ok with it, and let's deploy it

**Comment by rickybanister** [comment]
Yeah, let's get this test out there!
